File: numbers/src/main/java/org/dicio/numbers/formatter/param/NiceDurationParameters.java
package org.dicio.numbers.formatter.param;

import org.dicio.numbers.formatter.Formatter;
import org.dicio.numbers.unit.Duration;

public class NiceDurationParameters {
    // TODO documentation

    private final Formatter formatter;
    private final Duration duration;

    // default values
    private boolean speech = true;

    public NiceDurationParameters(final Formatter formatter, final Duration duration) {
        this.formatter = formatter;
        this.duration = duration;
    }

    public NiceDurationParameters speech(final boolean speech) {
        this.speech = speech;
        return this;
    }

    public String get() {
        return formatter.niceDuration(duration, speech);
    }
}

File: numbers/src/main/java/org/dicio/numbers/formatter/param/NiceDurationParameters.kt
package org.dicio.numbers.formatter.param

import org.dicio.numbers.formatter.Formatter
import org.dicio.numbers.unit.Duration

/**
 * TODO documentation
 */
class NiceDurationParameters(
    private val formatter: Formatter, private val duration: Duration
) {
    // default values
    private var speech = true

    fun speech(speech: Boolean): NiceDurationParameters {
        this.speech = speech
        return this
    }

    fun get(): String {
        return formatter.niceDuration(duration, speech)
    }
}
